mdx/granule_metadata_extractor/processing/process_hiwat.py

```diff
@@ -0,0 +1,135 @@
+from ..src.extract_netcdf_metadata import ExtractNetCDFMetadata
+import os
+import numpy as np
+from netCDF4 import Dataset
+from datetime import datetime, timedelta
+import pygrib
+
+class ExtractHiwatMetadata(ExtractNetCDFMetadata):
+    """
+    A class to extract HIWAT
+    """
+
+    def __init__(self, file_path):
+        self.file_path = file_path
+
+        filename = file_path.split('/')[-1]
+        if '_d02.grb2' in filename:
+           self.fileformat = 'GRIB2'
+           # extracting time and space metadata for GRIB2 file
+           [self.minTime, self.maxTime, self.SLat, self.NLat, self.WLon, self.ELon] = \
+               self.get_variables_min_max_grib2()
+        else: #netCDF-4
+           self.fileformat = 'netCDF-3'
+           # extracting time and space metadata for netCDF-3 file
+           [self.minTime, self.maxTime, self.SLat, self.NLat, self.WLon, self.ELon] = \
+               self.get_variables_min_max_netcdf()
+
+    def get_variables_min_max_netcdf(self):
+        """
+        :return:
+        """
+        """
+        Extract temporal and spatial metadata from netCDF files
+        """
+        # open nc file
+        nc = Dataset(self.file_path)
+
+        #Example file: 'wrfout_d02_2017-04-04_06-00-00'
+        #nc['XTIME'].units
+        #'minutes since 2017-04-02 18:00:00'
+
+        utc_minute = nc.variables['XTIME'][:]
+        time_att = nc.variables['XTIME'].getncattr('units')
+        utc_ref = datetime.strptime(time_att,'minutes since %Y-%m-%d %H:%M:%S')

```

Comment:
For future development we should avoid `datetime.strptime` as it is very inefficient. Instead you can use `.split()` and the date time constructor object directly for better performance.